api/src/test/java/org/pmiops/workbench/api/WorkspacesControllerTest.java

```diff
@@ -444,6 +480,31 @@ public void testCloneWorkspace() throws Exception {
     assertThat(workspace2.getUserRoles().get(0).getRole()).isEqualTo(WorkspaceAccessLevel.OWNER);
   }
 
+  @Test
+  public void testCloneWorkspaceWithCohorts() throws Exception {
+    Workspace workspace = createDefaultWorkspace();
+    workspace = workspacesController.createWorkspace(workspace).getBody();
+
+    Cohort c1 = createDefaultCohort(workspace, "c1");
+    c1 = cohortDao.save(c1);
+    Cohort c2 = createDefaultCohort(workspace, "c2");
+    c2 = cohortDao.save(c2);
+
+    stubGetWorkspace(workspace.getNamespace(), workspace.getName(),
+        LOGGED_IN_USER_EMAIL, WorkspaceAccessLevel.OWNER);
+    CloneWorkspaceRequest req = new CloneWorkspaceRequest();
+    Workspace modWorkspace = new Workspace();
+    modWorkspace.setName("cloned");
+    modWorkspace.setNamespace("cloned-ns");
+    ResearchPurpose modPurpose = new ResearchPurpose();
+    modPurpose.setAncestry(true);
+    modWorkspace.setResearchPurpose(modPurpose);
+    req.setWorkspace(modWorkspace);
+    workspacesController.cloneWorkspace(workspace.getNamespace(), workspace.getId(), req)
+        .getBody().getWorkspace();
+    verify(cohortService, times(2)).saveAndCloneReviews(any(), any());
```

Comment:
Seems like we should test that the cohorts got cloned here? (And reviews under them?)

Reply:
Partly done, see other comment on cohort reviews (still WIP).